### PR TITLE
boards: lock EUI provider to interface type

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Otherwise the provider can be used for multiple interfaces but only return a single MAC address for all of them.

### Testing procedure

Add `USEMODULE += dose` to `examples/gnrc_networking` when building for `same54-xpro`.

#### before

Both DOSE and Ethernet interface use the same MAC address from the at24mac chip.

```
2021-06-02 13:54:52,130 # Iface  6  HWaddr: FC:C2:3D:0D:2D:1F 
2021-06-02 13:54:52,134 #           L2-PDU:1500  MTU:1500  HL:64  RTR  
2021-06-02 13:54:52,136 #           RTR_ADV  
2021-06-02 13:54:52,139 #           Source address length: 6
2021-06-02 13:54:52,141 #           Link type: wired
2021-06-02 13:54:52,147 #           inet6 addr: fe80::fec2:3dff:fe0d:2d1f  scope: link  VAL
2021-06-02 13:54:52,150 #           inet6 group: ff02::2
2021-06-02 13:54:52,152 #           inet6 group: ff02::1
2021-06-02 13:54:52,156 #           inet6 group: ff02::1:ff0d:2d1f
2021-06-02 13:54:52,157 #           
2021-06-02 13:54:52,160 #           Statistics for Layer 2
2021-06-02 13:54:52,163 #             RX packets 0  bytes 0
2021-06-02 13:54:52,167 #             TX packets 1 (Multicast: 1)  bytes 78
2021-06-02 13:54:52,170 #             TX succeeded 1 errors 0
2021-06-02 13:54:52,173 #           Statistics for IPv6
2021-06-02 13:54:52,176 #             RX packets 0  bytes 0
2021-06-02 13:54:52,180 #             TX packets 1 (Multicast: 1)  bytes 64
2021-06-02 13:54:52,183 #             TX succeeded 1 errors 0
2021-06-02 13:54:52,183 # 
2021-06-02 13:54:52,187 # Iface  5  HWaddr: FC:C2:3D:0D:2D:1F 
2021-06-02 13:54:52,191 #           L2-PDU:1500  MTU:1500  HL:64  RTR  
2021-06-02 13:54:52,193 #           RTR_ADV  
2021-06-02 13:54:52,196 #           Source address length: 6
2021-06-02 13:54:52,198 #           Link type: wired
2021-06-02 13:54:52,204 #           inet6 addr: fe80::fec2:3dff:fe0d:2d1f  scope: link  TNT[1]
2021-06-02 13:54:52,207 #           inet6 group: ff02::2
2021-06-02 13:54:52,210 #           inet6 group: ff02::1
2021-06-02 13:54:52,213 #           inet6 group: ff02::1:ff0d:2d1f
2021-06-02 13:54:52,214 #           
2021-06-02 13:54:52,217 #           Statistics for Layer 2
2021-06-02 13:54:52,220 #             RX packets 0  bytes 0
2021-06-02 13:54:52,224 #             TX packets 1 (Multicast: 1)  bytes 0
2021-06-02 13:54:52,227 #             TX succeeded 0 errors 1
2021-06-02 13:54:52,230 #           Statistics for IPv6
2021-06-02 13:54:52,233 #             RX packets 0  bytes 0
2021-06-02 13:54:52,237 #             TX packets 1 (Multicast: 1)  bytes 64
2021-06-02 13:54:52,241 #             TX succeeded 1 errors 0
```

#### with this patch

The MAC address is only used once.

```
2021-06-02 14:03:50,929 # Iface  6  HWaddr: A6:ED:29:EC:CB:F3 
2021-06-02 14:03:50,933 #           L2-PDU:1500  MTU:1500  HL:64  RTR  
2021-06-02 14:03:50,935 #           RTR_ADV  
2021-06-02 14:03:50,938 #           Source address length: 6
2021-06-02 14:03:50,941 #           Link type: wired
2021-06-02 14:03:50,946 #           inet6 addr: fe80::a4ed:29ff:feec:cbf3  scope: link  VAL
2021-06-02 14:03:50,949 #           inet6 group: ff02::2
2021-06-02 14:03:50,952 #           inet6 group: ff02::1
2021-06-02 14:03:50,955 #           inet6 group: ff02::1:ffec:cbf3
2021-06-02 14:03:50,956 #           
2021-06-02 14:03:50,959 #           Statistics for Layer 2
2021-06-02 14:03:50,962 #             RX packets 0  bytes 0
2021-06-02 14:03:50,966 #             TX packets 1 (Multicast: 1)  bytes 78
2021-06-02 14:03:50,970 #             TX succeeded 1 errors 0
2021-06-02 14:03:50,972 #           Statistics for IPv6
2021-06-02 14:03:50,975 #             RX packets 0  bytes 0
2021-06-02 14:03:50,980 #             TX packets 1 (Multicast: 1)  bytes 64
2021-06-02 14:03:50,983 #             TX succeeded 1 errors 0
2021-06-02 14:03:50,983 # 
2021-06-02 14:03:50,986 # Iface  5  HWaddr: FC:C2:3D:0D:2D:1F 
2021-06-02 14:03:50,990 #           L2-PDU:1500  MTU:1500  HL:64  RTR  
2021-06-02 14:03:50,992 #           RTR_ADV  
2021-06-02 14:03:50,995 #           Source address length: 6
2021-06-02 14:03:50,998 #           Link type: wired
2021-06-02 14:03:51,003 #           inet6 addr: fe80::fec2:3dff:fe0d:2d1f  scope: link  VAL
2021-06-02 14:03:51,006 #           inet6 group: ff02::2
2021-06-02 14:03:51,009 #           inet6 group: ff02::1
2021-06-02 14:03:51,012 #           inet6 group: ff02::1:ff0d:2d1f
2021-06-02 14:03:51,013 #           
2021-06-02 14:03:51,016 #           Statistics for Layer 2
2021-06-02 14:03:51,019 #             RX packets 0  bytes 0
2021-06-02 14:03:51,023 #             TX packets 1 (Multicast: 1)  bytes 0
2021-06-02 14:03:51,027 #             TX succeeded 0 errors 1
2021-06-02 14:03:51,029 #           Statistics for IPv6
2021-06-02 14:03:51,032 #             RX packets 0  bytes 0
2021-06-02 14:03:51,037 #             TX packets 1 (Multicast: 1)  bytes 64
2021-06-02 14:03:51,040 #             TX succeeded 1 errors 0
```

### Issues/PRs references

alternative to #16521

### CODE REVIEW

**Review by fjmolinas — COMMENTED** [review]
If we want to go down that path I think that the default value should be overridable so add header guards and the header documentation should be added probably with an assert as well so that `NETDEV_ANY` is not used.

**Comment by benpicco** [comment]
I updated the documentation and added an `assert` to eui_provider.c.

File: boards/same54-xpro/include/eui_provider_params.h [hunk 1]
```diff
@@ -38,6 +38,7 @@ static inline int _at24mac_get_eui48(uint8_t index, eui48_t *addr)
  * @{
  */
 #define EUI48_PROVIDER_FUNC   _at24mac_get_eui48
+#define EUI48_PROVIDER_TYPE   NETDEV_SAM0_ETH
```
Comment (fjmolinas) [inline boards/same54-xpro/include/eui_provider_params.h:41]:
```suggestion
#ifndef EUI48_PROVIDER_TYPE
#define EUI48_PROVIDER_TYPE   NETDEV_SAM0_ETH
#endif
```

File: boards/native/include/eui_provider_params.h [hunk 2]
```diff
@@ -30,7 +30,7 @@ extern "C" {
  * @{
  */
 #define EUI64_PROVIDER_FUNC   native_cli_get_eui64
-#define EUI64_PROVIDER_TYPE   NETDEV_ANY
+#define EUI64_PROVIDER_TYPE   NETDEV_SOCKET_ZEP
```
Comment (fjmolinas) [inline boards/native/include/eui_provider_params.h:33]:
```suggestion
```suggestion
#ifndef EUI64_PROVIDER_TYPE
#define EUI64_PROVIDER_TYPE   NETDEV_SOCKET_ZEP
#endif
```
```

**Pull request merged**